moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_planning.cpp

```diff
@@ -242,6 +242,16 @@ void MotionPlanningFrame::onFinishedExecution(bool success)
   // update query start state to current if neccessary
   if (ui_->start_state_combo_box->currentText() == "<current>")
     startStateTextChanged(ui_->start_state_combo_box->currentText());
+
+  // update previous state if plan is successfully executed
+  if (success)
```

Comment:
Why only doing this on success?

Reply:
I supposed to keep a goal state In failure case. But it should be done with "previous goal state" (currently not exists). I will change to update "previous state" on both of success and failure.

Reply:
The most common execution failure I see is the drivers not being connected and the robot not moving. I don't know that there is a case where the execution fails and the previous state should be overwritten anyway.

Reply:
Whenever the user presses `Execute`, they basically committed to this next motion and likely do not care about the previous one anymore.

You might very well get cases where the execution aborts mid-trajectory, e.g., through MoveIt's monitoring mechanisms and controller safety features.

Reply:
> The most common execution failure I see is the drivers not being connected.

Another common case is that execution fails due to trajectory tracking issues in the middle of the motion.